python/dgl/graphbolt/internal/sample_utils.py

```diff
@@ -294,12 +294,29 @@ def unique_and_compact_csc_formats(
     return unique_nodes, compacted_csc_formats
 
 
+def _broadcast_timestamps(csc, dst_timestamps):
+    count = torch.diff(csc.indptr)
+    src_timestamps = torch.repeat_interleave(dst_timestamps, count)
+    return src_timestamps
+
+
 def compact_csc_format(
     csc_formats: Union[CSCFormatBase, Dict[str, CSCFormatBase]],
     dst_nodes: Union[torch.Tensor, Dict[str, torch.Tensor]],
+    dst_timestamps: Optional[
+        Union[torch.Tensor, Dict[str, torch.Tensor]]
+    ] = None,
 ):
     """
-    Compact csc formats and return original_row_ids (per type).
+    Relabel the row (source) IDs in the csc formats into a contiguous range from
+    0 and return the original row node IDs per type.
+
+    Note that
+    1. The column (destination) IDs are included in the relabeled row IDs.
+    2. If there are repeated row IDs, they would not be uniqued and will be
+    treated as different nodes.
+    3. If `dst_timestamps` is given, the timestamp of each destination node will
```

Comment:
Is this by design? That is, source nodes' timestamps is determined by corresponding destination nodes?

Reply:
Yes. The the source nodes are used as the frontier nodes in the next layer. They should retain the timestamps of the frontier nodes in this layer to filter neighbors based on the timestamps of the seed nodes.